composer.json

```diff
@@ -12,6 +12,9 @@
 	"require": {
 		"php": ">=7.0"
 	},
+	"suggest": {
+		"ext-dom": "*"
```

Comment:
I could have put this in `require` since the extension is required when running tests (and not just the tests being introduced in this PR). But since it is not required to actually run WordPress, I went with `suggest`.

Reply:
PhpStorm, at least, rightly identifies this as having been missing:

![image](https://github.com/WordPress/wordpress-develop/assets/134745/1290a666-343a-4cf8-ba3d-afe5baa2bef7)

Reply:
Another reason to probably not use this approach and stick to the simple `str_replace` that doesn't rely on new dependencies :)